Documentation/In-depth/ComposerJson/Index.rst

```diff
@@ -0,0 +1,285 @@
+         "typo3/cms-form": "^10.4.15",
+         "typo3/cms-frontend": "^10.4.15",
+         "typo3/cms-impexp": "^10.4.15",
+         "typo3/cms-info": "^10.4.15",
+         "typo3/cms-install": "^10.4.15",
+         "typo3/cms-recordlist": "^10.4.15",
+         "typo3/cms-rte-ckeditor": "^10.4.15",
+         "typo3/cms-seo": "^10.4.15",
+         "typo3/cms-setup": "^10.4.15",
+         "typo3/cms-sys-note": "^10.4.15",
+         "typo3/cms-t3editor": "^10.4.15",
+         "typo3/cms-tstemplate": "^10.4.15",
+         "typo3/cms-viewpage": "^10.4.15"
+      },
+      "autoload": {
+         "psr-4": {},
+         "classmap": []
+      },
+      "config": {
+         "sort-packages": true
+      },
+      "extra": {
+         "typo3/cms": {
+            "web-dir": "public"
+         }
+      },
+      "scripts": {
+         "typo3-cms-scripts": [
+            "typo3cms install:generatepackagestates",
+            "typo3cms install:fixfolderstructure"
+         ],
+         "post-autoload-dump": [
+            "@typo3-cms-scripts"
+         ]
+      }
+   }
+
+You may notice that we include the third party extension
+`helhum/typo3-console` here. This is recommended and the
+section `scripts` uses it (see below).
+
+Commands for creating composer.json
+===================================
+
+It is possible to create a composer.json from scratch with
+`composer init`. However, in practice, it may be a good idea
+to have a general composer.json which you reuse for every project
+and add individual requirements for your projects via command line
+commands.
+
+In general it is recommended to use the command line to modify
+composer.json (as in `composer require ...`). If you modify
+a :file:`composer.json` in an existing project directly, you must
+execute a command like `composer install` or `composer require`
+afterwards to make sure that your :file:`composer.lock` will be updated.
+
+(Incomplete) example for creating above :file:`composer.json`
+(scripts is missing):
+
+.. code-block:: shell
+
+   composer init --type project; \
+   composer require \
+   helhum/typo3-console:^6.4.0; \
+   typo3/cms-backend:^10.4.15; \
+   typo3/cms-belog:^10.4.15; \
+   typo3/cms-beuser:^10.4.15; \
+   typo3/cms-core:^10.4.15; \
+   typo3/cms-dashboard:^10.4.15; \
+   typo3/cms-extbase:^10.4.15; \
+   typo3/cms-extensionmanager:^10.4.15; \
+   typo3/cms-felogin:^10.4.15; \
+   typo3/cms-filelist:^10.4.15; \
+   typo3/cms-fluid:^10.4.15; \
+   typo3/cms-fluid-styled-content:^10.4.15; \
+   typo3/cms-form:^10.4.15; \
+   typo3/cms-frontend:^10.4.15; \
+   typo3/cms-impexp:^10.4.15; \
+   typo3/cms-info:^10.4.15; \
+   typo3/cms-install:^10.4.15; \
+   typo3/cms-recordlist:^10.4.15; \
+   typo3/cms-rte-ckeditor:^10.4.15; \
+   typo3/cms-seo:^10.4.15; \
+   typo3/cms-setup:^10.4.15; \
+   typo3/cms-sys-note:^10.4.15; \
+   typo3/cms-t3editor:^10.4.15; \
+   typo3/cms-tstemplate:^10.4.15; \
+   typo3/cms-viewpage:^10.4.15; \
+    \
```

Comment:
The backslash here and and all following must be removed, otherwise all of this is part of the previous `require` command.

Reply:
There was a semicolon at the end. But there are some problems here anyway as too many semicolons etc. will fix.